cpp/ql/src/Security/CWE/CWE-120/OverrunWrite.ql

```diff
@@ -21,12 +21,12 @@ import semmle.code.cpp.commons.Alloc
  * See CWE-120/UnboundedWrite.ql for a summary of CWE-120 alert cases.
  */
 
-from BufferWrite bw, Expr dest, int destSize, int estimated
+from BufferWrite bw, Expr dest, int destSize, int estimated, TypeBoundsAnalysis reason
```

Comment:
We need to make sure that we don't report any alerts twice with these two queries (i.e., there's no result for which `bw.getMaxDataLimited(any(TypeBoundsAnalysis reason))` and `bw.getMaxDataLimited(any(ValueFlowAnalysis reason))` holds). That sounds like something that's true. Is that correct?

Reply:
yes. Apart from making sure the code had this property, in a previous version of my other PR where the reasons where printed I double checked there were no duplicate reasons being reported for the same cases. I think the DCA experiment that @geoffw0 run had that reason printing in, so I'm pretty confident there is no overlap. I'll still check that further though.